### PR TITLE
feat: add custom log levels

### PR DESCRIPTION
**Pull request by dima-bzz**

### CODE REVIEW

**Comment by dima-bzz** [comment]
@sevdokimov Can I leave only the default logging levels and let the rest be added by the users themselves if necessary?

File: log-viewer/src/main/java/com/logviewer/data2/LogLevels.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // See org.apache.log4j.lf5.LogLevel, org.apache.log4j.Level, https://en.wikipedia.org/wiki/Syslog
+    private static final String[] LEVELS = {
+            "OFF", "FATAL", "SEVERE", "EMERGENCY", "ALERT", "CRITICAL", "ERROR",
+            "WARN", "WARNING",
+            "INFO", "CONFIG", "NOTICE", "INFORMATIONAL",
+            "DEBUG", "FINE", "FINER",
+            "FINEST", "TRACE", "ALL",
+    };
+
+    private static String[] CUSTOM_LEVELS = new String[0];
```
Comment (sevdokimov) [inline log-viewer/src/main/java/com/logviewer/data2/LogLevels.java:19]: I don't like using a static field to store a level list. I'll try to invent something else. If I can't create another solution, I'll merge the code that uses the static field.

**Comment by dima-bzz** [comment]
@sevdokimov 👌

**Comment by sevdokimov** [comment]
> @sevdokimov Can I leave only the default logging levels and let the rest be added by the users themselves if necessary?

Which code piece do you mean?

**Comment by dima-bzz** [comment]
@sevdokimov By default, leave only Fatal, Error, Warn, Warning, Info, Debug and Trace, and let the users add all the other levels themselves.

**Comment by dima-bzz** [comment]
@sevdokimov Remove all other levels and put them in the config file.

**Comment by sevdokimov** [comment]
> @sevdokimov Remove all other levels and put them in the config file.

People don't like to customize configuration, it would be nice if everything work out of the box. Adding non-standard levels to `config.conf` helps when the user uses log-viewer as a standalone application, but when log-viewer is embedded into another application as a library, no `config.conf` file is used.

**Comment by dima-bzz** [comment]
@sevdokimov I understood.

**Comment by dima-bzz** [comment]
@sevdokimov I haven't reached the embedded application yet.

**Review by sevdokimov — APPROVED** [review]

File: log-viewer/src/main/java/com/logviewer/data2/LogLevels.java (hunk 1 above)
Comment (sevdokimov) [reply log-viewer/src/main/java/com/logviewer/data2/LogLevels.java:19]: I've created `List<String> customLevels` field in AbstractPatternLogFormat. Custom fields can be specified using
```
logs = [
  {
    path: ${HOME}"/my-app/logs/*.log"
    format: {
      type: LogbackLogFormat
      customLevels: ["XXX", "YYY"]
      pattern: "%date{yyyy-MM-dd_HH:mm:ss.SSS} [%thread] %-5level %logger{35} - %X{pipelineId}%X{contentId}%msg%n"
    }
  }
]
```
or
```
log-viewer.log-levels=["TEST"]
```

It's important to have the level list in `LogFormat` object when log-viewer works in multinode mode. The configuration may be different on different nodes, but `LogFormat` will be same, log-viewer sends it to other nodes when it needs to load data from other node.

Comment (dima-bzz) [reply log-viewer/src/main/java/com/logviewer/data2/LogLevels.java:19]: @sevdokimov Great, I've been thinking about that too.

**Pull request merged**